src/Client.ts

```diff
@@ -344,53 +356,88 @@ export default class Client extends EventEmitter {
             this.transport.disconnect(false);
         }
 
-        const transportPref = ['websocket', 'bosh'];
-        let endpoints: { [key: string]: string[] } | undefined;
+        const transportPref = this.config.transportPreferenceOrder ?? [];
+        const transportEndpoints: Array<[Transport, TransportConfig]> = [];
+        let endpoints: { [key: string]: string[] } = {};
+        try {
+            endpoints = await (this as unknown as Agent).discoverBindings(
+                this.config.server!
+            );
+        } catch (e) {
+            console.error(e);
```

Comment:
What's the error here? Does `discoverBindings()` send out an error event? How's the user know we failed here?

Reply:
Failing `discoverBindings` here doesn't matter. It can fail for example if host-meta or host-meta.json don't exist, in which case it will fall back either to TCP or whatever other endpoint it knows about (e.g. from the user manually specifying one).